### PR TITLE
Updating example for MultiChildLayoutDelegate

### PR DESCRIPTION
**Pull request by gspencergoog**
This makes the example that MultiChildLayoutDelegate be compiled with the rest of the example code, and forms a more complete example.

Also, added some extra debug output for a couple of asserts.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 1]
```diff
@@ -41,26 +41,42 @@ class MultiChildLayoutParentData extends ContainerBoxParentData<RenderBox> {
...
+///   follower,
+/// }
+/// 
+/// class FollowTheLeader extends MultiChildLayoutDelegate {
+///   @override
+///   void performLayout(Size size) {
+///     Size followerSize = Size.zero;
+/// 
+///     if (hasChild(_Slot.leader)) {
+///       followerSize = layoutChild(_Slot.leader, new BoxConstraints.loose(size));
```
Comment (xqwzts) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: Would it make more sense for this var to be named `leaderSize`?
- `layoutChild` returns the size of the widget it processes.
- `follower` is constrained/positioned according to the `leader`'s size.

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 2]
```diff
@@ -41,26 +41,42 @@ class MultiChildLayoutParentData extends ContainerBoxParentData<RenderBox> {
 /// Used with [CustomMultiChildLayout], the widget for the
 /// [RenderCustomMultiChildLayoutBox] render object.
 ///
-/// ## Example
+/// Each child must be wrapped in a [LayoutId] widget to assign the id that
+/// identifies it to the delegate.
+///
+/// ## Example code
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: you want "Sample code" iirc

File: packages/flutter/lib/src/widgets/framework.dart [hunk 3]
```diff
@@ -4462,13 +4462,17 @@ abstract class RenderObjectElement extends Element {
   @override
   void deactivate() {
     super.deactivate();
-    assert(!renderObject.attached);
+    assert(!renderObject.attached,
+      'A RenderObject was still attached when attempting to deactivate its '
+      'RenderObjectElement: ${renderObject.toString()}');
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: the toString is implied

File: packages/flutter/lib/src/widgets/framework.dart [hunk 4]
```diff
@@ -4462,13 +4462,17 @@ abstract class RenderObjectElement extends Element {
...
   }
 
   @override
   void unmount() {
     super.unmount();
-    assert(!renderObject.attached);
+    assert(!renderObject.attached,
+      'A RenderObject was still attached when attempting to unmount its '
+      'RenderObjectElement: ${renderObject.toString()}');
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: ditto

File: packages/flutter/lib/src/rendering/custom_layout.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/rendering/custom_layout.dart]: i agree that leaderSize makes more sense

File: packages/flutter/lib/src/rendering/custom_layout.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/rendering/custom_layout.dart]: Done.

File: packages/flutter/lib/src/rendering/custom_layout.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/rendering/custom_layout.dart]: OK, done.

File: packages/flutter/lib/src/widgets/framework.dart (hunk 3 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/framework.dart]: Duh.  I had toStringDeep there before, and wasn't thinking when I just removed the "Deep".  Fixed.

**Comment by gspencergoog** [comment]
PTAL: I think this is ready to commit.

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 5]
```diff
@@ -41,26 +41,42 @@ class MultiChildLayoutParentData extends ContainerBoxParentData<RenderBox> {
 /// Used with [CustomMultiChildLayout], the widget for the
 /// [RenderCustomMultiChildLayoutBox] render object.
 ///
-/// ## Example
+/// Each child must be wrapped in a [LayoutId] widget to assign the id that
```
Comment (xster) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: While you're here, should we add a comment on the `id`'s uniqueness requirements here and in CustomMultiChildLayout

**Review by xster — APPROVED** [review]
LG

File: packages/flutter/lib/src/rendering/custom_layout.dart (hunk 5 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/rendering/custom_layout.dart]: Done, and in the docs for LayoutId.

**Pull request merged**